File: app/src/main/java/com/example/final_pocproject/Activity/MainActivity.java
package com.example.final_pocproject.Activity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Patterns;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.example.final_pocproject.Retrofit.ApiInterface;
import com.example.final_pocproject.R;
import com.example.final_pocproject.Model.User;
import com.example.final_pocproject.Retrofit.RetrofitClient;
import com.google.android.material.textfield.TextInputLayout;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {


    private Button loginButton;
    private TextInputLayout textInputEmail;
    private ApiInterface apiInterface;




    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        loginButton = findViewById(R.id.button);
        textInputEmail = findViewById(R.id.email);

        apiInterface = RetrofitClient.getRetrofit().create(ApiInterface.class);

      loginButton.setOnClickListener(new View.OnClickListener() {
          @Override
          public void onClick(View v) {

              if(validateEmail()){
                  getUsers();
              }
          }
      });

    }

    private boolean validateEmail(){
        String inputEmail = textInputEmail.getEditText().getText().toString().trim();

        if(inputEmail.isEmpty()){
            textInputEmail.setError("Email field can't be empty");
            return false;
        }else if (!Patterns.EMAIL_ADDRESS.matcher(inputEmail).matches()){
            textInputEmail.setError("Please enter a valid email id");
            return false;
        }else {
            textInputEmail.setError(null);
            return true;
        }
    }

    private  void getUsers(){
        Call<List<User>> call = apiInterface.getUsers();
        call.enqueue(new Callback<List<User>>() {
            @Override
            public void onResponse(Call<List<User>> call, Response<List<User>> response) {
                if(!response.isSuccessful()){
                    return;
                }
                 List<User> users = response.body();
                String email = textInputEmail.getEditText().getText().toString();


                int userId = 0;
                for (User user : users){
                    if(user.getEmail().equals(email)){
                        userId=user.getId();

                        break;
                    }
                }

                if(userId==0){
                    textInputEmail.setError("Email id not found");
                }else {
                    Intent intent = new Intent(MainActivity.this, PostActivity.class);
                    intent.putExtra("id",userId);
                    startActivity(intent);
                }
            }
            @Override
            public void onFailure(Call<List<User>> call, Throwable t) {

                Toast.makeText(MainActivity.this,t.getMessage(),Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: app/src/main/java/com/example/pocProjectKotlin/activities/MainActivity.kt
package com.example.pocProjectKotlin.activities

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import com.example.pocProjectKotlin.*
import com.example.pocProjectKotlin.interfaces.UserInterface
import com.example.pocProjectKotlin.model.User
import com.example.pocProjectKotlin.api.UserApi
import com.example.pocProjectKotlin.sessionmanagement.SessionManager
import com.google.android.material.textfield.TextInputLayout


class MainActivity : AppCompatActivity(), UserInterface {

    private lateinit var loginButton: Button
    private lateinit var inputEmail: TextInputLayout
    private val emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+"
    private lateinit var userApi: UserApi
    private lateinit var sessionManager: SessionManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        sessionManager = SessionManager(applicationContext)
        if (sessionManager.isLoggedIn()) {
            val intent = Intent(this, DashboardActivity::class.java)
            finish()
            startActivity(intent)
        }

        loginButton = findViewById(R.id.loginButton)
        inputEmail = findViewById(R.id.email)
        userApi = UserApi(this)

        loginButton.setOnClickListener(View.OnClickListener {

            val emailId: String = inputEmail.editText?.text.toString().trim()

            if (validEmail(emailId))
                userApi.getUser()
        })
    }


    private fun validEmail(emailId: String): Boolean {
        var isValid = false
        if (emailId.isEmpty()) {
            inputEmail.error = getString(R.string.emailErrorMessage)
        } else if (emailId.matches(emailPattern.toRegex())) {
            inputEmail.error = null
            isValid = true
        } else {
            inputEmail.error = getString(R.string.invalidEmail)
        }
        return isValid
    }

    override fun handleSuccessResponse(users: List<User>) {
        val emailEntered: String = inputEmail.editText?.text.toString()
        var userId: Int = 0
        for (i: Int in users.indices)
            if (users[i].email.equals(emailEntered, true)) {
                userId = users[i].id
            }

        if (userId != 0) {
            sessionManager.createLoginSession(emailEntered, userId)
            val intent = Intent(this@MainActivity, DashboardActivity::class.java)
            finish()
            startActivity(intent)
        } else {
            inputEmail.error = getString(R.string.emailNotFound)
        }
    }

    override fun handleFailure(t: Throwable) {
        Toast.makeText(this@MainActivity, t.message, Toast.LENGTH_SHORT).show()
    }

    override fun responseNotStressful(responseCode: Int) {
        Toast.makeText(this@MainActivity, """Code : $responseCode""", Toast.LENGTH_SHORT).show()
    }
}

